### DISCUSSION TITLE
Coordinate system of OBJ model

### DISCUSSION THREAD

**Discussion by informatter** [alicevision/Meshroom#1693] [Q&A]
Hi guys, 

I am interested in knowing the coordinate system of the vertices in the .obj file after the scene is reconstructed, is it in ECEF XYZ coordinates?

Thanks

**Comment by natowi** [comment]
Meshroom uses its own local coordinate system:

![grafik](https://user-images.githubusercontent.com/8138537/168661781-f73c1a81-427f-4ee4-a1ef-4d394f64681b.png)

If you did not use cctags for orientation or manually rotated your model via the gui and the sfm transform node, your model will be placed more or less in a "random" orientation in the scene. This orientation will be based on the initial pair, which may change from one run to another. You can align the model using the sfm transform node and "from camera" for example to align the angle.
Also see https://github.com/alicevision/meshroom/wiki/CCTAG-scaling

**Reply by informatter** [reply]
Hi @natowi 
Thanks for your reply. Ok I see, so Meshroom uses a Y-up left-handed coordinate system then? Also, I am not sure what you mean by _"This orientation will be based on the initial pair"_ could you please clarify this?

I am actually running Meshroom on a server, so I am just using the CLI interface.  I am checking the docs for the sfm transform node https://meshroom-manual.readthedocs.io/en/bibtex1/node-reference/nodes/SfMTransform.html so it seems that If I choose `auto_from_cameras` as the transform method, the point clouds X,Y,Z would be aligned with the + Y axis, and the model would no longer be in a random orientation?

Following this line of thought, another question arises. I don't see that the sfm transform method is exposed in the CLI. I am currently just using `meshroom_batch` How can I use that node from the command line, is it exposed somewhere as an executable? also, I am guessing that I would need to create my own  `meshroom_batch` pipeline which includes the usage of the sfm transform, as it's not being used in `meshroom_batch` 

My end goal is to transform the .obj file from Meshroom's coordinate system to ECEF https://en.wikipedia.org/wiki/Earth-centered,_Earth-fixed_coordinate_system#/media/File:Ecef_coordinates.svg. Because that is the coordinate system I need the mesh to be in. This would be the last step after the correct transformation is applied by using the  sfm transform node.

Sorry for all the questions, I am actually a new user of Meshroom and I have only used it from the CLI ! 

Thank you for your time, it's really appreciated.

**Reply by natowi** [reply]
On each run, Meshroom picks an initial image pair to start the 3D reconstruction with. When I remember correctly, a camera is picked from this, set in origin and used for orientation. Since the selection of the initial pair can change from each run, the orientation can vary on each run.
For that reason, there is the SfMTransform node. It can be use to set a specific camera (image) as origin. If you capture large scenes, I can recommend you to start your capturing session with an aligned image (perfectly horizontal). With from_single_camera this image can be used to align the model. You can use the file name or viewID:
![grafik](https://user-images.githubusercontent.com/8138537/169894242-145ee608-a147-4a9f-a057-c0c149ce354b.png)

To automatically scale your model, use cctags as linked above.

You could write your own cli script or create a .mg file in the gui and adjust the paths later: https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/cmd/compute.html
